### PR TITLE
openfoam v1906 reorganized their wmake rules and now rely on General/…

### PR DESCRIPTION
**Pull request by mboisson**
…*/{c,c++} for compiler parameters

### CODE REVIEW

File: easybuild/easyblocks/o/openfoam.py [hunk 1]
```diff
@@ -181,6 +181,8 @@ def configure_step(self):
 
         # inject compiler variables into wmake/rules files
         ldirs = glob.glob(os.path.join(self.builddir, self.openfoamdir, 'wmake', 'rules', 'linux*'))
+        if LooseVersion(self.version) >= LooseVersion('v1906'):
```
Comment (boegel) [inline easybuild/easyblocks/o/openfoam.py]: This comparison is only meaningful if the version is also of the form `v[0-9]`, so make sure that's the case?

```python
if self.version.startswith('v') and LooseVersion(self.version) >= LooseVersion('v1906'):
```

File: easybuild/easyblocks/o/openfoam.py [hunk 2]
```diff
@@ -210,6 +212,9 @@ def configure_step(self):
             'c++OPT': os.environ['CXXFLAGS'],
         }
         for wmake_rules_file in wmake_rules_files:
+            # the cOpt and c++Opt files don't exist in the General directories
```
Comment (boegel) [inline easybuild/easyblocks/o/openfoam.py]: should be reworded?

```python
# the cOpt and c++Opt files don't exist in the General directories (which are included for recent versions)

File: easybuild/easyblocks/o/openfoam.py (hunk 1 above)
Comment (mboisson) [reply easybuild/easyblocks/o/openfoam.py]: done

File: easybuild/easyblocks/o/openfoam.py (hunk 2 above)
Comment (mboisson) [reply easybuild/easyblocks/o/openfoam.py]: done

File: easybuild/easyblocks/o/openfoam.py [hunk 3]
```diff
@@ -181,6 +181,8 @@ def configure_step(self):
 
         # inject compiler variables into wmake/rules files
         ldirs = glob.glob(os.path.join(self.builddir, self.openfoamdir, 'wmake', 'rules', 'linux*'))
+        if self.version.startswith('v') and LooseVersion(self.version) >= LooseVersion('v1906'):
```
Comment (boegel) [inline easybuild/easyblocks/o/openfoam.py]: @mboisson please sync with current `develop` and use `self.looseversion > LooseVersion('1906'):`, see changes in #1794

Comment (mboisson) [reply easybuild/easyblocks/o/openfoam.py]: Done

**Comment by akesandgren** [comment]
@boegel are you satisified with the fix to your request? Then please handle it.

**Comment by boegel** [comment]
@mboisson Blocked by conflicts with current `develop`...

**Comment by mboisson** [comment]
@boegel it should be merged now.

**Comment by akesandgren** [comment]
Going in, thanks @mboisson!

**Pull request merged**